packages/@lwc/engine-core/src/framework/renderer.ts

```diff
@@ -60,7 +60,11 @@ export interface RendererAPI {
     getLastElementChild: (element: E) => E | null;
     getTagName: (element: E) => string;
     isConnected: (node: N) => boolean;
-    insertStylesheet: (content: string, target?: ShadowRoot) => void;
+    insertStylesheet: (
+        content: string,
+        target: ShadowRoot | undefined,
+        signal: AbortSignal | undefined
+    ) => void;
```

Comment:
I decided to use an [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) because I think it's a nice way to have a clean separation between `engine-core` and `engine-dom`. `engine-core` only ever needs to deal with the `AbortController`, and `engine-dom` only ever needs to deal with the `AbortSignal`. `engine-core` is responsible for triggering the abort, and `engine-dom` just needs to respond to the `abort` event.

In the future, you could imagine abstracting this in such a way that each `AbortController` is tied to an individual `StylesheetFactory` rather than just a string; that might help with implementing #4155.